packages/react-reconciler/src/ReactFiberLazyComponent.js

```diff
@@ -50,9 +50,9 @@ export function readLazyComponentType<T>(lazyComponent: LazyComponent<T>): T {
       thenable.then(
         moduleObject => {
           if (lazyComponent._status === Pending) {
-            const defaultExport = moduleObject.default;
+            const _result = moduleObject ? moduleObject.default : moduleObject;
```

Comment:
```suggestion
            const _result = moduleObject != null ? moduleObject.default : moduleObject;
```
It's probably a bit exotic to check for strings or numbers but it's closer to the original behavior.